### PR TITLE
docs: upgrade contributing.md guide

### PR DESCRIPTION
**Pull request by vladimir-shirmanov**
added more instructions for future contributors, to make the process smoother

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,3 +12,23 @@ The base version represents the MAJOR and MINOR parts of [SemVer](https://semver
...
+  - fix: A bug fix
+  - docs: Documentation only changes
+  - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+  - refactor: A code change that neither fixes a bug nor adds a feature
+  - perf: A code change that improves performance
+  - test: Adding missing tests or correcting existing tests
+  - build: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
+  - ci: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
+  - chore: Other changes that don't modify src or test files
+  - revert: Reverts a previous commit
```
Comment (vbreuss) [inline CONTRIBUTING.md:30]: Please check the formatting in the preview window. I would suggest the following changes:
```md
Please use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to name your commits and PR title.

Available prefixes:
  - `feat:`  
    A new feature
  - `fix:`  
     A bug fix
  - `docs:`  
     Documentation only changes
  - `style:`  
     Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
  - `refactor:`  
     A code change that neither fixes a bug nor adds a feature
  - `perf:`  
     A code change that improves performance
  - `test:`  
     Adding missing tests or correcting existing tests
  - `build:`  
     Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
  - `ci:`  
     Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
  - `chore:`  
     Other changes that don't modify src or test files
  - `revert:`  
     Reverts a previous commit
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,3 +12,23 @@ The base version represents the MAJOR and MINOR parts of [SemVer](https://semver
 
 - `main` contains the latest sources. Each merge there triggers a deploy to `nuget.org`.
 - All versions on `nuget.org` have a matching GitHub release/tag.
+
+## Commits and PR title
+
+- Please use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to name your commits and PR title
+  Available types:
```
Comment (vbreuss) [inline CONTRIBUTING.md:19]: I think "available prefixes" would be more clear.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,3 +12,23 @@ The base version represents the MAJOR and MINOR parts of [SemVer](https://semver
...
+  - docs: Documentation only changes
+  - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+  - refactor: A code change that neither fixes a bug nor adds a feature
+  - perf: A code change that improves performance
+  - test: Adding missing tests or correcting existing tests
+  - build: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
+  - ci: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
+  - chore: Other changes that don't modify src or test files
+  - revert: Reverts a previous commit
+- Examples:
```
Comment (vbreuss) [inline CONTRIBUTING.md:31]: I think examples are not necessary here. Closed pull requests can serve as a template and adding some here will just make this file unnecessary long.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,3 +12,23 @@ The base version represents the MAJOR and MINOR parts of [SemVer](https://semver
 
 - `main` contains the latest sources. Each merge there triggers a deploy to `nuget.org`.
 - All versions on `nuget.org` have a matching GitHub release/tag.
+
+## Commits and PR title
+
+- Please use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to name your commits and PR title
```
Comment (vbreuss) [inline CONTRIBUTING.md:18]: As we use [amannn/action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#configuration) for enforcing this policy, maybe we could add a link here?

File: CONTRIBUTING.md (hunk 1 above)
Comment (vladimir-shirmanov) [reply CONTRIBUTING.md]: yeah, this way it will be much clearer. Fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (vladimir-shirmanov) [reply CONTRIBUTING.md]: sure thing. Fixed

File: CONTRIBUTING.md (hunk 3 above)
Comment (vladimir-shirmanov) [reply CONTRIBUTING.md]: Agree, removed them from the file

File: CONTRIBUTING.md (hunk 4 above)
Comment (vladimir-shirmanov) [reply CONTRIBUTING.md]: Yes, that's a good additional source of information for future contributors

**Review by vbreuss — APPROVED** [review]
Thanks a lot!

**Pull request merged**